### PR TITLE
[MGDOBR-888] Add E2E test for bridge Error Handler update

### PR DESCRIPTION
**Pull request by Srihari1192**
[MGDOBR-888](https://issues.redhat.com/browse/MGDOBR-888)

Note: Waiting for PR https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/875 to merge

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/BridgeSteps.java [hunk 1]
```diff
@@ -111,6 +112,24 @@ public void createFakeBridge(String testBridgeName) {
         bridgeContext.setDeleted(true);
     }
 
+    @When("^update the Bridge \"([^\"]*)\" with body:$")
+    public void updateBridge(String testBridgeName, String bridgeRequestJson) {
+        BridgeContext bridgeContext = context.getBridge(testBridgeName);
+        String systemBridgeName = BridgeResource.getBridgeDetails(context.getManagerToken(), bridgeContext.getId()).getName();
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/BridgeSteps.java:118]: AFAIK you can get Bridge system name using `bridgeContext.getName()`. Can you try it?

Comment (Srihari1192) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/BridgeSteps.java:118]: Yes it worked , Thanks

**Review by jiripetrlik — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

**Comment by sutaakar** [comment]
Waiting for https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/875, right?
Please add the link and note in the PR description. Also add `on hold` label.

**Comment by jomarko** [comment]
@Srihari1192 thank you for your response. Now, I understand the error is in sent `data` where we have `myMessage` instead of expected `sender` specified in transformation template.

However to me it seems we are still sending cloud event directly to error handler. I think so especially because of this line:
```
When send a cloud event to the Ingress of the Bridge "ehBridgeUpdate":
```

Please correct me if I am worng, shouldn't the above be rather:
```
When send a cloud event to the Ingress of the Bridge "whProcessor":
```
?

**Comment by manstis** [comment]
I think you want to rebase this or something?

**Comment by Srihari1192** [comment]
> I think you want to rebase this or something?

yes I tried to do rebase ..got some error reverted back

**Comment by manstis** [comment]
#875 has been merged :fireworks: 

I don't suppose I could interest you in adding a test for [this](https://issues.redhat.com/browse/MGDOBR-975) too?

**Comment by jiripetrlik** [comment]
Good point @manstis . I think it would be possible to modify this test which is changing error handler "from webhook to webhook" to "from webhook to kafka" scenario. This will mean that we also test the Kafka error handler. As I have some experience with Kafka action integration test, I've created a JIRA for it here https://issues.redhat.com/browse/MGDOBR-994 .

**Pull request merged**